### PR TITLE
try pre-commit add pyupgrade

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
adding pyupgrade and pre-commit to clean some cruft

### CODE REVIEW

**Comment by ThomasWaldmann** [comment]
This PR has a lot of unrelated changes, can you reduce it to what you intentionally wanted to change?

**Comment by RonnyPfannschmidt** [comment]
@ThomasWaldmann this is a sideeffect of pre-commit actions, i'll reiterate the active hooks as a starting point as this adds to much as of now

File: scripts/glibc_check.py [hunk 1]
```diff
@@ -35,12 +35,12 @@ def main():
             output = subprocess.check_output(objdump % filename, shell=True,
                                              stderr=subprocess.STDOUT)
             output = output.decode()
-            versions = set(parse_version(match.group(1))
-                           for match in glibc_re.finditer(output))
+            versions = {parse_version(match.group(1))
+                           for match in glibc_re.finditer(output)}
```
Comment (ThomasWaldmann) [inline scripts/glibc_check.py:39]: hmm, do we really want to *fix* such stuff in pre-commit, trusting that it always is correct and acceptig that it does not fix the indentation?

File: src/borg/algorithms/zstd/lib/common/zstd_internal.h [hunk 2]
```diff
@@ -261,7 +261,7 @@ typedef enum {
  *         - ZSTD_overlap_src_before_dst: The src and dst may overlap, but they MUST be at least 8 bytes apart.
  *           The src buffer must be before the dst buffer.
  */
-MEM_STATIC FORCE_INLINE_ATTR 
+MEM_STATIC FORCE_INLINE_ATTR
```
Comment (ThomasWaldmann) [inline src/borg/algorithms/zstd/lib/common/zstd_internal.h:264]: src/borg/algorithms/zstd/* and some other dirs there is not our code, but 3rd party code we just bundle - we should not modify it for cosmetics.

File: scripts/glibc_check.py (hunk 1 above)
Comment (RonnyPfannschmidt) [reply scripts/glibc_check.py:39]: Typically it's used in conjunction with the flake8 and the black hook, but that's way too much at once

I don't consider it always right, but it is a massive help for finger memory sediments

File: src/borg/algorithms/zstd/lib/common/zstd_internal.h (hunk 2 above)
Comment (RonnyPfannschmidt) [reply src/borg/algorithms/zstd/lib/common/zstd_internal.h:264]: Good catch, i need to exclude them

**Review by ThomasWaldmann — COMMENTED** [review]
there is also a test.txt in this commit.

**Comment by ThomasWaldmann** [comment]
what do we do with this one?

**Comment by RonnyPfannschmidt** [comment]
closing for now, will revisit this mid next year at earliest

**Pull request closed without merging**